packages/react-magma-dom/src/components/Popover/PopoverContent.tsx

```diff
@@ -0,0 +1,224 @@
+  /**
+   * @internal
+   */
+  testId?: string;
+  /**
+   * @children required
+   */
+  children: React.ReactChild | React.ReactChild[];
+}
+
+const StyledCard = styled(Card)<{
+  position: PopoverPosition;
+  isInverse?: boolean;
+  isOpen?: boolean;
+  maxHeight?: string;
+  width?: string;
+  theme?: ThemeInterface;
+  hasPointer?: boolean;
+}>`
+  position: relative;
+  background: ${props =>
+    props.isInverse
+      ? props.theme.colors.primary600
+      : props.theme.colors.neutral100};
+  display: ${props => (props.isOpen ? 'block' : 'none')};
+  max-height: ${props => (props.maxHeight ? props.maxHeight : '100%')};
+  opacity: ${props => (props.isOpen ? '1' : '0')};
+  outline: 0;
+  transition: opacity 0.3s;
+  white-space: nowrap;
+  border: 1px solid
+    ${props =>
+      props.isInverse
+        ? props.theme.colors.primary400
+        : props.theme.colors.neutral300};
+  width: ${props => (props.width ? props.width : '100%')};
+
+  ::before {
+    display: ${props => (props.hasPointer ? 'block' : 'none')};
+    content: '';
+    position: absolute;
+    width: 0;
+    height: 0;
+    border-style: solid;
+    border-width: 11px;
+  }
+
+  &[data-popover-placement=${PopoverPosition.bottom}]::before {
+    border-color: transparent transparent
+      ${props =>
+        props.isInverse
+          ? props.theme.colors.primary400
+          : props.theme.colors.neutral300}
+      transparent;
+    left: 50%;
+    bottom: 100%;
+    transform: translateX(-50%) rotate(0deg);
+  }
+  &[data-popover-placement=${PopoverPosition.top}]::before {
+    border-color: ${props =>
+        props.isInverse
+          ? props.theme.colors.primary400
+          : props.theme.colors.neutral300}
+      transparent transparent transparent;
+    left: 50%;
+    top: 100%;
+    transform: translateX(-50%) rotate(0deg);
+  }
+
+  &::after {
+    display: ${props => (props.hasPointer ? 'block' : 'none')};
+    content: '';
+    position: absolute;
+    width: 0;
+    height: 0;
+    border-style: solid;
+    border-width: 10px;
+  }
+
+  &[data-popover-placement=${PopoverPosition.bottom}]::after {
+    border-color: transparent transparent
+      ${props =>
+        props.isInverse
+          ? props.theme.colors.primary600
+          : props.theme.colors.neutral100}
+      transparent;
+    left: 50%;
+    bottom: 100%;
+    transform: translateX(-50%) rotate(0deg);
+  }
+  &[data-popover-placement=${PopoverPosition.top}]::after {
+    border-color: ${props =>
+        props.isInverse
+          ? props.theme.colors.primary600
+          : props.theme.colors.neutral100}
+      transparent transparent transparent;
+    left: 50%;
+    top: 100%;
+    transform: translateX(-50%) rotate(0deg);
+  }
+`;
+
+const StyledAnnounce = styled(Announce)`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+`;
+
+const ErrorMessage = styled.div`
+  width: fit-content;
+  padding: 1em;
+`;
+
+const StyledPopoverContent = styled.div`
+  z-index: 2;
```

Comment:
I haven't pulled the code but this reminds me, how does the popover behave/interact with other elements with z-indexes? What if it's inside a drawer or next to a dropdown?